src/components/poll/create/SearchOptionModal.tsx

```diff
@@ -21,6 +21,10 @@ const isMedia = (obj: any): obj is Anilist.Media => Boolean(obj && obj.title);
 const isCharacter = (obj: any): obj is Anilist.Character =>
   Boolean(obj && obj.name);
 
+const mediaName = (obj: any): obj is Anilist.Media => {
```

Comment:
or should i have actually taken in Anilist.Media as a type specifically instead of any?

Reply:
What it's happening on this `isCharacter` function is:
It's takes an object of any type `obj: any` and the return type is `obj is Anilist.Character`, this `obj is Anilist.Character` as return type is a bit confusing but yeah, there i have to explicit the return type because if a let the compiler decides it will set the return as a boolean since `Boolean(obj && obj.name)` is a boolean.

So in your code you don't want to return if the obj is a media you want to return the media title which is a string (but you don't need to explicit that, since the compiler can identify that by the return) so would be something like: 
```ts
const mediaName = (obj: Anilist.Media) => {
     return obj.title.english ?? obj.title.romaji ?? obj.title.native;
};
```
i would prefer something like:
```ts
const getMediaName = (media: Anilist.Media) => media.title.english ?? media.title.romaji ?? media.title.native;
```

Reply:
thats what i was going to do originally but then decided that your way was probably better lol. ill fix and commit again